src/lib.rs

```diff
@@ -151,40 +152,80 @@ struct HelperInner {
     consumer_done: bool,
 }
 
-/// Error type for `from_env` function.
+/// Return type for `from_env_ext` function.
 #[derive(Debug)]
-pub enum ErrFromEnv {
-    /// There isn't env var, that describes jobserver to inherit.
-    IsNotConfigured,
-    /// Cannot connect following this process's environment.
-    PlatformSpecific {
-        /// Error.
-        err: io::Error,
-        /// Name of gotten env var.
-        env: &'static str,
-        /// Value of gotten env var.
-        var: String,
-    },
+pub struct FromEnv {
+    /// Result of trying to get jobserver client from env.
+    pub client: Result<Client, FromEnvError>,
+    /// Name of the environment variable.
+    /// Empty if no relevant environment variable is found.
+    pub var_name: &'static str,
+    /// Value of the environment variable.
+    /// Empty if no relevant environment variable is found.
+    pub var_value: OsString,
```

Comment:
Why not use `Option` to represent no env variable is found and group them together as one field?
```suggestion
    /// Name and value of the environment variable.
    /// `None` if no relevant environment variable is found.
    pub var: Option<(&'static str, OsString)>,
```

Reply:
`FromEnvError::NoEnvVar` also indicates that there's no env var and these fields can be ignored.
If there's also an `Option` they can in theory go into conflict, and you'll have to unwrap, so I guess no `Option` is simpler.

("Empty if no relevant environment ..." could easily be "Unspecified if no relevant environment ...")

Reply:
Maybe it's better to use Result<(Client, EnvVariable), FromEnvError> and also includes EnvVariable in FromEnvError except for variant NoEmvVar and NotSupporte?

    struct EnvVariable {
        name: &'static str,
        value: OsString,
    }

Reply:
In case of negative file descriptor `from_env_ext` may want to return `Client` without environment variables. Maybe it's better to make result and env independent like `(Result<Client, FromEnvError>, Option<EnvVariable>)`?

Reply:
> In case of negative file descriptor `from_env_ext` may want to return `Client` without environment variables.

I don't `from_env_ext` should return Client on negative file descriptor.

Negative file descriptor means the jobserver is disabled.

In that case, the user could create a client using Client::nee